hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusDeleteTopicHandler.java

```diff
@@ -66,6 +66,7 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         final var op = context.body().consensusDeleteTopicOrThrow();
         final var topicStore = context.createStore(ReadableTopicStore.class);
         // The topic ID must be present on the transaction and the topic must exist.
+        mustExist(op.topicID(), INVALID_TOPIC_ID);
```

Comment:
If op.topicID is null, topic will be null on line 70 because we used TopicID.DEFAULT instead. You will get INVALID_TOPIC_ID.

Reply:
But we should fail early and not look at state with `TopicID.DEFAULT` if `TopicID` is not set.  But I agree it would have already failed in `pureChecks`

Reply:
Interesting point, @iwsimon. I will resolve here to be able to merge. Please continue the discussion here: #12811